ts/src/config.ts

```diff
@@ -148,6 +162,8 @@ export interface ProfilerConfig extends GoogleAuthOptions {
   localProfilingPeriodMillis: number;
   localLogPeriodMillis: number;
   localTimeDurationMillis: number;
+  sourceMapSearchPath: string[];
```

Comment:
This is WIP.

Reply:
Can you just delete the sourceMapPaths field from ProfilerConfig? Or it won't build then?

Reply:
It doesn't build how? I.e. what are the errors?

Making the field optional still feels like a massage rather than a fix. Conceptually, you have a thing A here that is used to produce thing B as part of initialization and the thing B is what you feed into the profiler class. Still carrying thing A as part of that input to the profiler is redundant / confusing.

Reply:
I don't want to trigger a major refactoring though, so I mostly wonder if there is an easy / elegant way to address this.

Reply:
I don't think there's a really simple way to change this. Here are the solutions I can think of. All would require a bit of refactoring of the current code, but I don't think any would be really major.

Option 1:

`initLocalConfig` could return a different type (which I'd call `LocalConfig`, possibly). 

Option 1A:
`ProfilerConfig` would have all current fields (fields present before any source map support), plus the following:
```
export interface ProfilerConfig extends GoogleAuthOptions {
  ...
  sourceMapPath: string[];
  disableSourceMaps: boolean;
}
```

`LocalConfig` would be unexported, and defined in index.ts. `LocalConfig` would have all fields that ProfilerConfig in current source code has, plus the following:
```
interface LocalConfig extends GoogleAuthOptions {
  ...
  sourceMapSearchPath: string[];
  disableSourceMaps: boolean;
}
```

Option 1B:
Roughly the same as 1A, but while we currently initialize the SourceMapper when calling `profiler.start()`, for this option, the initialized SourceMapper would be part of the ProfilerConfig (and the config would be initialized in `initConfigAsync`). 

`ProfilerConfig` would have all current fields, plus the following:
```
export interface ProfilerConfig extends GoogleAuthOptions {
  ...
  sourceMapper?: SourceMapper;
}
```
Source map support would be disabled when the sourceMapper is undefined.

`LocalConfig` would be  the same as in 1A.


Option 2:
Right now, we find source maps files during config initialization, and initialize the SourceMapper when we call `profiler.start()`.

We could shift finding source map files given `sourceMapSearchPath` to when we call `profiler.start()`

Then `ProfilerConfig` would have all current fields (fields present before any source map support), plus the following:
```
export interface ProfilerConfig extends GoogleAuthOptions {
  ...
  sourceMapSearchPath: string[];
  disableSourceMaps: boolean;
}
```

If none of these options are good or you think a hybrid approach might be better, let me know, and I can move this to a design document --I think the structure of a design document might be better for discussions where we're considering multiple options.

I have some preference for options 1B and 2, because with those options group all of the work done to initialize the SourceMapper into one place. With either of these options, we could have `sourceMapSearchPath`  be an argument for `create` (in sourcemapper.ts) instead of `sourceMapPaths`. 

I think I could implement option 2 with minimal changes, so I'll go ahead and do that.

Reply:
Implemented option 2.

Reply:
Implemented option 2.

Reply:
Option 2 SGTM as far as I can tell from the description. There might be some wrinkles in terms of it being asynchronous and error checking but hopefully not. Let's see if we can settle the change here, a design doc may be an overkill.

Reply:
Marking this as resolved for now -- we've settled on a general approach and the approach is now implemented. So we can make more specific comments about that approach in the code.